### PR TITLE
refactor install_sublime_text.ps1

### PR DESCRIPTION
**Pull request by guillermooo**
@randy3k This is ready for merging. Let me know if you want me to squash the commits.

### CODE REVIEW

**Comment by randy3k** [comment]
I am curious why you keep refactoring the scripts? Coz most people do not care how they works at all 🤨

**Comment by guillermooo** [comment]
> I am curious what make you keep refactoring the scripts? Coz most people do not care how they works at all 🤨

Several reasons, really, for example:

- They don't stop on the first error
- They lack documentation
- Often there are magic numbers
- Sometimes we are not using the simplest PS solution to a problem (for example, extracting a zip archive)
- We are using PS defaults explicitly at times (the script would work just as well without them)
- Logging can be improved
- Some aspects could be made configurable (for example, hard-coded paths for ST installation)

So basically, to clean up. But once you start doing that you realize there's code that could be reused in different places, etc.

If you are not comfortable with my changes, feel free to say so! I think it's better to unify the style in all PS scripts and try to reuse as much code as possible. But I can also propose changes before actually making them so you can approve them first.

Also, ultimately, I would like to make the PS scripts cross platform so they can be used on Linux, macOS and Windows.

**Comment by randy3k** [comment]
Thanks for the amazing work. I didn't know that Powershell has ported to linux (sorry for my ignorance). Then I understand why you spent so much effort in refactoring the scripts :)

Yup, most of the CI scripts are poorly written/documented because they were written at a time that I didn't expect to make UnitTesting public. I was also too lazy to rewrite them after UnitTesting went public as they kind of work.

Anyway, I appreciate your time in it. Too sleepy to review this now, will take a look tomorrow.

**Comment by guillermooo** [comment]
Also, I am making extensive use of UnitTesting in some projects, so I benefit directly from reusable code, etc.
